### DISCUSSION TITLE
Setting xl = xu for some parameters to set them

### DISCUSSION THREAD

**Discussion by MidnightBiscuit** [anyoptimization/pymoo#512] [Q&A]
Hi everyone,
I have a problem with 15 parameters and want to test different configurations. Sometimes I want to optimise only using two free parameters, sometimes four ou six etcetc. The others have to be set by me. To keep some flexibility I defined the problem with `n_var=15` and change the boundaries to reflect which parameters are free and which are not. For instance a problem with 4 free parameters and 11 parameters set to a given value looks like this
```python
    xl=np.array([1, 1, 1, 1, 0.5, 0, 0.5,
                 1, 1, 1,    0.5, 0, 0.5,
                 80,40]),                   
    xu=np.array([ 1, 1, 1, 1, 2, 0,  2,  
                  1, 1, 1,    2, 0,  2,   
                 80,40]), 
```
My question is : is this a valid way of doing it ? Does the algorithm manage this in a nice way or should I arange my program to only include genuine free parameters in the boundaries ? I don't want the program to do something crazy becase I suggest that I have 15 free parameters but then set some of them with a degenerated boundaries.
I tested this and the result _seems_ to be reasonable, no errors, result has a physical meaning and the solution is in the expected range. But maybe it is not optimum or a bit unwise ?

For my problem I use the following algorithm and minimisation routine
```python
algorithm = NSGA2(
    pop_size=20,
    n_offsprings=10,
    sampling=FloatRandomSampling(),
    crossover=SBX(prob=0.8, eta=10),
    mutation=PM(eta=20),
    eliminate_duplicates=True
)

res = minimize(problem,
               algorithm,
               seed=1,
               save_history=False,
               verbose=True
)

```
Problem is inherited from `ElementwiseProblem`, I have two objectives, zero constraints.

**Comment by blankjul** [comment]
I would recommend always defining a new problem where only the free parameters are set (for some algorithms defining a variable with `xl = xu` might work but then for others not. I think I remember that it worked for NSGA2 with its default settings. If there are no errors then there should not be any side-effects and you can simplify it like you do.
